pylint/checkers/refactoring.py

```diff
@@ -369,6 +372,52 @@ def visit_boolop(self, node):
                              node=node,
                              args=(duplicated_name, ', '.join(names)))
 
+    @utils.check_messages('consider-using-ternary')
+    def visit_assign(self, node):
+        if self._is_and_or_ternary(node.value):
+            cond, truth_value, false_value = self._get_and_or_ternary_arguments(node.value)
+        elif self._is_seq_based_ternary(node.value):
+            cond, truth_value, false_value = self._get_seq_based_ternary_params(node.value)
+        else:
+            return
+        self.add_message('consider-using-ternary', node=node,
+                         args=(truth_value.as_string(), cond.as_string(), false_value.as_string()),)
+
+    visit_return = visit_assign
+
+    @staticmethod
+    def _is_and_or_ternary(node):
+        """
+        Returns true if node is 'condition and true_value else false_value' form.
+
+        All of: condition, true_value and false_value should not be a complex boolean expression
+        """
+        return (isinstance(node, astroid.BoolOp)
+                and node.op == 'or' and len(node.values) == 2
+                and isinstance(node.values[0], astroid.BoolOp)
+                and not isinstance(node.values[1], astroid.BoolOp)
+                and node.values[0].op == 'and'
+                and not isinstance(node.values[0].values[1], astroid.BoolOp))
+
+    @staticmethod
+    def _get_and_or_ternary_arguments(node):
+        false_value = node.values[1]
+        condition, true_value = node.values[0].values
```

Comment:
@rogalski 

Using a sentence like as: `my_var = True and True and True or True`
Show the following traceback:
```bash
Traceback (most recent call last):
...
  File "pylint/checkers/refactoring.py", line 400, in visit_assign
    cond, truth_value, false_value = self._get_and_or_ternary_arguments(node.value)
  File "pylint/checkers/refactoring.py", line 427, in _get_and_or_ternary_arguments
    condition, true_value = node.values[0].values
ValueError: too many values to unpack
```